### PR TITLE
Update composer spec

### PR DESCRIPTION
**Pull request by guzzilar**
> This PR requires PR #32 to be merged first.

---
> Note: This PR is a part of steps to make Omise-Magento be able to upload to the Magento Marketplace (extension store)
> - [x] Relocate extension files from Omise/Payment out to root dir (PR #31)
> - [x] Remove existing Omise-PHP from the project (switch to composer) (PR #32)
> - [x] Update composer spec (PR #33)
> - [ ] Merge all PRs back to `2-0-update-code-to-conform-with-magento-extension-guide`
> - [ ] Merge `2-0-update-code-to-conform-with-magento-extension-guide` to `2-1-0`

#### 1. Objective

Continue from #32 

To make Omise-Magento be able to submit to Magento Marketplace (extension store), we have to add PSR-4 spec to `composer.json`

And bonus point, this PR would make Omise-Magento be able to be installed through `composer` command.

**Related information**:
Related issue(s): 🙅
Related ticket(s): T2121

#### 2. Description of change

• Change package name in `composer.json` from `omise/payment` to `omise/omise-magento`

**why?**
To make it conform to all of Omise packages (that prefix name with `omise/omise-*`).

And also, because we're going to register this repository to [https://packagist.org](Packagist) that already have `omise/omise-php` package https://packagist.org/packages/omise/omise-php

So, the package that Im gonna register is something like `https://packagist.org/packages/omise/omise-magento`

• Add PSR-4 spec, because Magento store require that we have to specify "PSR-4" spec in composer.json.

• Specify Omise-PHP version to 2.6.0 to solve `..but these conflict with your requirements or minimum-stability` [issue](https://github.com/composer/composer/issues/5211) 

ref. http://stackoverflow.com/a/21467982

#### 3. Quality assurance

**🔧 Environments:**

- **Platform version**: Magento CE 2.1.2.
- **Omise plugin version**: 2.0 (2.1.0 now)
- **PHP version**: 7.0.14.

**✏️ Details:**

Tested by re-build Magento 2.1.2 system from scratch.
Then, run `composer` to install Omise-Magento.

1. `composer require omise/omise-magento:dev-2-0-update-composer-spec`.
  screen shot 2559-12-21 at 5 54 14 pm

2. Then, check `vendor` folder to make sure that Omise package be installed properly.
  screen shot 2559-12-21 at 5 55 34 pm

3. Run `php bin/magento module:enable Omise_Payment` and `php bin/magento setup:upgrade` to enable Omise module & register module to Magento system. 
  screen shot 2559-12-21 at 6 05 40 pm

4. Enter to admin page, at the page `STORES > Configuration > Advance > Advance`, you would see `Omise_Payment` module's config there.
  screen shot 2559-12-21 at 6 08 31 pm

5. Test config Omise module from `STORES > Configuration > Sales > Payment Method`.
  screen shot 2559-12-21 at 6 09 37 pm

6. Then, create new product and test checkout, you would see "Omise payment method" at the checkout form.
  screen shot 2559-12-21 at 7 00 59 pm

  screen shot 2559-12-21 at 7 02 46 pm

7. Back to admin page and check from the order list page, new order record should be appeared there.
  screen shot 2559-12-21 at 7 05 25 pm

  screen shot 2559-12-21 at 7 02 59 pm

8. Test make charge (checkout) again, but this time, set config to "Authorize & Capture".
screen shot 2559-12-21 at 7 11 35 pm

screen shot 2559-12-21 at 7 11 00 pm

#### 4. Impact of the change

- This change would make Omise-Magento repository be able to register to [https://packagist.org](Packagist) service and can be installed through `composer` command.

- Solved charge service issue that PR #32 made.

- Be able to submit to Magento Marketplace (extension store).

#### 5. Priority of change

Normal

#### 6. Additional Notes

No.

### CODE REVIEW

**Comment by nimid** [comment]
@guzzilar I have used a [tool](https://github.com/magento-ecg/coding-standard) to check the source code. The tool found 1 error and 1 warning. The warning can be ignored.

The screenshot below shows the error from checking by tool.
ecgm2_ _-bash_ _181x45

The error in the screenshot is the first error from the Magento Marketplace report.

The screenshot below shows the first error from the Magento Marketplace report.
magento_marketplace

To make sure that when submit the source code to Magento Marketplace the result is pass, I think this error must be solved.

I encourage to use the tool coupled with manually checking. The tool can helps to check the source code faster. The tool, the manual checking and the automatic checking can help to reduce the problems.

Note:
I use 2 tools. The first tool is the tool that I have mention in the discussion. The second tool is the tool above.

**Comment by guzzilar** [comment]
@nimid Thanks for that point 👍 . 
I just found this tool https://github.com/magento/marketplace-eqp seems more official (and still active, last updated is 7 days ago). 

-after run test, found the same issue-
screen shot 2559-12-22 at 12 30 46 pm

Btw, issue noted. 
Let me create another PR to solve this issue, cuz this one is about fix bug issue that not related directly with this PR (and need to have new test for that `exception` condition).

**Comment by guzzilar** [comment]
@nimid done here, https://github.com/omise/omise-magento/pull/34

**Comment by nimid** [comment]
@guzzilar 
> fix bug issue

What is the bug or issue you mention?

**Comment by guzzilar** [comment]
@nimid Here https://github.com/omise/omise-magento/pull/34

**Comment by guzzilar** [comment]
@nimid sry, I moved to #35 
(For the reason, please check my comment at #34)

Thanks

File: composer.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "type": "magento2-module",
     "license": "MIT",
     "require": {
-        "omise/omise-php": "dev-master"
```
Comment (nimid) [inline composer.json]: @guzzilar What is the reason to specific the version of a dependency, Omise-PHP, to 2.6.0?

Comment (guzzilar) [inline composer.json]: First, I got an error like, https://github.com/composer/composer/issues/5211
Then, I followed this suggestion http://stackoverflow.com/questions/21052831/how-to-solve-two-packages-requirements-conflicts-when-running-composer-install/21467982#21467982

**Comment by nimid** [comment]
@guzzilar I executed a command `php bin/magento system:upgrade` and got an error

```
[InvalidArgumentException]
There are no commands defined in the "system" namespace.
```

I think the command to upgrade the system after install the module is
`php bin/magento setup:upgrade`.

**Comment by nimid** [comment]
@guzzilar You can merge this pull request.

**Comment by guzzilar** [comment]
> I think the command to upgrade the system after install the module is
> php bin/magento setup:upgrade.

@nimid Oh, you were right, t's typo, I'll update in PR description.

**Comment by guzzilar** [comment]
👍  (and updated PR description)

**Pull request merged**